wooz1ewu/ITechArt_Front-End
# Note del/add, filters, infinite scroll, dnd, tests

# Overview 
> Notes adding/deleting implemented. Filter/search systems implemented, infinite scroll added. Dnd implemented. Sharing system added. Tests provided.

# Changes  
> 1. NotePanel component added.  
> 1. AddNotePanel, FilterByDatePanel, FilterByNamePanel components added.  
> 1. InfiniteScroll component added.  
> 1. Logout button moved to PageLayout component.  
> 1. LocalStorage methods added.  
> 1. Dnd functionality added to Note - components of NoteList.  
> 1. Notes sharing functionality implemented.  
> 1. Util tests added.  
> 1. Component tests added.  

# How to test

> 1. Visit [note-app page](https://wooz1ewu.github.io/ITechArt_Front-End/).  
> 1. Select the note you want to delete. Try to delete it by clicking on trash can icon.  
> 1. Try to add new note by clicking on plus icon.  
> 1. Try to add new note using default values (empty fields).  
> 1. Try to filter notes by date/name using panel (click on corresponding icons to call panel).  
> 1. Add some notes (8-10) and check the work of InfiniteScroll.  
> 1. Try to change notes order using drag-and-drop.  
> 1. Try to change notes order after applying some filters.  
> 1. Click on preferable note and try to share it to other system users.  
> 1. Check if shared note is displayed correctly.  
> 1. Unshare any shared note.
